packages/account/src/BiconomySmartAccountV2.ts

```diff
@@ -81,43 +81,123 @@ export class BiconomySmartAccountV2 extends BaseSmartAccount {
     this.nodeClient = new NodeClient({ txServiceUrl: nodeClientUrl ?? NODE_CLIENT_URL })
   }
 
-  async _getAccountContract(): Promise<SmartAccount_v100> {
+  async _getAccountContract(): Promise<SmartAccount_v200> {
     if (this.accountContract == null) {
-      this.accountContract = SmartAccountFactory_v100__factory.connect(
+      this.accountContract = SmartAccount_v200__factory.connect(
         await this.getAccountAddress(),
         this.provider
       )
     }
     return this.accountContract
   }
 
+  /**
+   * @description This function will initialise BiconomyAccount class state
+   * @returns Promise<BiconomyAccountV2>
+   */
+  /*async init(initilizationData?: InitializeV2Data): Promise<this> {
+    try {
+      let _accountIndex
+      if (initilizationData) {
+        _accountIndex = initilizationData.accountIndex
+      }
+
+      if (!_accountIndex) _accountIndex = 0
+      this.isProviderDefined()
+
+      await this._getAccountContract()
+
+      this.chainId = await this.provider.getNetwork().then((net) => net.chainId)
+      await this.initializeAccountAtIndex(_accountIndex)
+      this._isInitialised = true
+    } catch (error) {
+      Logger.error(`Failed to call init: ${error}`)
+      throw error
+    }
+
+    return this
+  }
+
+  async initializeAccountAtIndex(accountIndex: number): Promise<void> {
+    this.index = accountIndex
+    // TODO // Review should use index
+    this.accountAddress = await this.getAccountAddress()
+    // May not need at all
+    // await this.setContractsState()
+  }
+
+  private isInitialized(): boolean {
+    if (!this._isInitialised)
+      throw new Error(
+        'BiconomySmartAccountV2 is not initialized. Please call init() on BiconomySmartAccountV2 instance before interacting with any other functions'
+      )
+    return true
+  }*/
+
   // TODO
   async getNonce(): Promise<BigNumber> {
     if (await this.isAccountDeployed(await this.getAccountAddress())) {
       const accountContract = await this._getAccountContract()
-      return await accountContract.getNonce(0)
+      return await accountContract.nonce()
     }
     return BigNumber.from(0)
   }
 
+  // Review
+  // Overridden this method because entryPoint.callStatic.getSenderAddress() based on initCode doesnt always work 
+  // in case of account is deployed you would get AA13 or AA10 
+
+  /**
+   * calculate the account address even before it is deployed
+   */
+  async getCounterFactualAddress(): Promise<string> {
```

Comment:
yes, yesterday I realised I missed the index in getAccountAddress() and getCounterFactualAddress().

it's useful since now we rely on querying the factory.getAddressForCounterFactualAccount() in this child class. 

in the base class, it is still await this.entryPoint.callStatic.getSenderAddress(initCode) but I faced some issues with this as if account gets deployed once this method starts returning 0 address for counterfactual if I use initCode for already deployed SA. Not sure how it works in infinitism sdk.

Reply:
done above change